services/search/testing/test_search_engine.py

```diff
@@ -1,5 +1,70 @@
+from unittest import mock
 from search_engine import SearchEngine  # pylint: disable=import-error
 import test_constants as constants
+import utils  # pylint: disable=import-error
+
+
+def reqs_list(arr):
+    """
+    Returns a list of same length as arr
+    """
+    return [""] * len(arr)
+
+
+def to_json(req):
+    """
+    Returns the mocked json response with ten synonyms returned
+    """
+    return {"sinonimos": [{"sinonimo": "my_synonym"}] * constants.TEN_SYNONYMS}
+
+
+def test_one_synonym():
+    """
+    Tests one synonym call with default max synonyms (5)
+    """
+    result = ["my_synonym"] * constants.DEFAULT_MAX_SYNONYMS
+
+    with mock.patch('utils.SynonymExtractor.create_conc_reqs', side_effect=reqs_list):
+        with mock.patch('utils.SynonymExtractor.req_to_json', side_effect=to_json):
```

Comment:
Hmm, I would have expected just one mock for `self.session.get()`. I think if you mock out that response, you don't need to mock out anything else in `SynonymExtractor`.

Reply:
I thought so too and it should be possible, but the response object seems to be different than a normal response, where a normal response can be accessed using only `.json()` v.s. here you need to use `.result().json()` I did spend some time looking for the response object in the requests_futures repo and couldn't find it, this issue was discussed with @ssmall on our 1:1 coding session and this is the solution he proposed to me.

Reply:
But with your other suggestion this is possible

Reply:
I see, there's technically like two for-loops going untested here, which isn't ideal.

Because python is weakly typed and just keeps trying to execute lines until it gets an error, I believe you can actually use a mock class defined in just the test file to simulate an object with `.result().json()` and return that as the mock of `self.session.get()`. It would take two nested classes though.

Something like:
```python
class MockJson:
    __init__(self, json_result_value_you_want_to_use_for_test):
         self.json = json_result_value_you_want_to_use_for_test
    def json(self):
         return self.json

class MockResult:
    __init__(self, json_result_value_you_want_to_use_for_test):
         self.mock_json = new MockJson(json_result_value_you_want_to_use_for_test)
    def result(self):
        return self.mock_json

def test():
    req_list = [new MockResult('{"some": "json"}'), new MockResult('{"some": "json"})]
    with mock.patch('self.session.get', side_effect=reqs_list)
```